projects/js-packages/publicize-components/src/hooks/use-media-restrictions/index.js

```diff
@@ -31,13 +33,22 @@ const getImageValidationError = ( sizeInMb, maxImageSize ) =>
  *
  * @param {number} sizeInMb - The fileSize in bytes.
  * @param {number} length - Video length in seconds and.
+ * @param {number} width - Width of the video.
+ * @param {number} height - Height of the video.
  * @param {object} videoLimits - Has the properties to check against
  * @returns {(FILE_SIZE_ERROR | VIDEO_LENGTH_TOO_LONG_ERROR | VIDEO_LENGTH_TOO_SHORT_ERROR)} Returns validation error.
  */
-const getVideoValidationError = ( sizeInMb, length, videoLimits ) => {
-	const { minSize, maxSize, minLength, maxLength } = videoLimits;
-
-	if ( ! sizeInMb || sizeInMb > maxSize || sizeInMb < minSize ) {
+const getVideoValidationError = ( sizeInMb, length, width, height, videoLimits ) => {
+	const {
+		minSize = 0,
+		maxSize = GLOBAL_MAX_SIZE,
+		minLength = 0,
+		maxLength = GLOBAL_MAX_SIZE,
+		maxWidth = GLOBAL_MAX_SIZE,
+		aspectRatio = DEFAULT_RESTRICTIONS.video.aspectRatio,
+	} = videoLimits;
+
+	if ( ! sizeInMb || sizeInMb > maxSize || sizeInMb < minSize || width > maxWidth ) {
```

Comment:
As discussed, maybe the width restriction should be grouped with the aspect ratio, and be a DIMENSIONS_ERROR or something